course/steps/08_react-apollo/pt.md

```diff
@@ -0,0 +1,82 @@
+
+# Conectando *back-end* e *front-end*
+
+## Introdução
+
+Agora aprenderemos como recuperar dados do *backend* e exibí-los na interface. O VTEX IO utiliza [GraphQL](https://graphql.org/) como linguagem/tecnologia para transferência de dados, o que torna a programação dos nossos componentes bastante simples. Iremos modificar o nosso componente Countdown para buscar o *targetDate* do **campo `releaseDate` de um produto da VTEX**. Para realizar queries GraphQL em React, é utilizado o **Apollo Client**, uma biblioteca de gerenciamento de estado que facilita a integração de uma API GraphQL com a aplicação *front-end*.
+
+A biblioteca **Apollo Client** disponibiliza uma integração nativa com React, por meio de *hooks*. Dessa forma, realizar uma *query* significa usar um *hook* que não só realizará as *queries* e fará o *fetch* dos dados, mas também proverá cache e atualização do estado da UI. Essa integração, chamada `react-apollo` já está declarada no `package.json`.
+
+## Preparação
+- Para implementar esta funcionalidade, precisamos **adicionar o nosso bloco `countdown` na página de produto**, e também faremos nossos testes nessa página também. Para isso, faça o seguinte:
+1. Em seu tema clonado (`store-theme`) acesse o arquivo `store/blocks/product.jsonc` e, no bloco `flex-layout.col#right-col` adicione o bloco `countdown`, logo antes do `buy-button`:
+	```diff
+	    "product-gifts",
+	+	"countdown",
+	    "flex-layout.row#buy-button",
+	    "availability-subscriber",
+	```
+2. Rode `vtex link` em seu tema novamente (caso o processo já não esteja sendo executado).
+3. Pronto, agora o nosso bloco está na página de produto. Acesse alguma destas páginas e veja o componente `Countdown` renderizado.
+
+## Query de Release Date
+
+1. Crie uma pasta `react/queries` e nela adicione um arquivo `productReleaseDate.graphqql` que irá conter a *query* a ser feita. Em particular, essa *query* irá receber um termo, que será **o slug do produto a ser recuperado a data de lançamento**. Ela chamará o *resolver* `product`, já disponível pela app `vtex.search-graphql`, e recuperaremos apenas o campo que precisamos.
+    ```
+    query productReleaseDate($slug: String){
+		  product(slug: $slug) {
```

Comment:
maybe decrease tab spacing

Reply:
It's necessary to not create another numbered list (after 1.)

Reply:
actually I realized that I didn't explain it properly, it was the tab spacing inside the query (line 27), but it's better now